src/sources/http_server.rs

```diff
@@ -364,6 +380,9 @@ impl SourceConfig for SimpleHttpConfig {
 
         let source = SimpleHttpSource {
             headers: build_param_matcher(&remove_duplicates(self.headers.clone(), "headers"))?,
+            custom_response_headers: self.custom_response_headers.clone().into_iter().map(|(k, v)| {
+                (k.parse().unwrap(), v.into_iter().map(|v| v.parse().unwrap()).collect())
```

Comment:
Rather than `unwrap` can we propagate up any header parsing errors as errors here? Currently Vector would panic at start-up if there was an error parsing the headers.